client/src/pages/SignUp.js

```diff
@@ -0,0 +1,243 @@
+  },
+  textField: {
+    margin: theme.spacing(2),
+    width: "30ch",
+  },
+  whiteText: {
+    color: "white",
+  },
+  marginLeft_2: {
+    margin: "2em",
+  },
+});
+
+class SignUpPage extends Component {
+  state = {
+    username: "",
+    email: "",
+    password: "",
+    usernameError: "",
+    emailError: "",
+    passwordError: "",
+    passwordConfirmError: "",
+  };
+
+  validateChange = (e) => {
+    switch (e.target.name) {
+      case "signupName": {
+        if (e.target.value.length === 0) {
+          this.setState({ usernameError: "Field should not be empty." });
+        } else {
+          this.setState({ usernameError: "" });
+          this.setState({ username: e.target.value });
+        }
+        break;
+      }
+      case "signupEmail": {
+        if (
+          e.target.value.split("").filter((x) => x === "@").length !== 1 ||
+          e.target.value.indexOf(".") === -1
+        ) {
+          this.setState({ emailError: "Not a valid email id." });
+        } else {
+          this.setState({ emailError: "" });
+          this.setState({ email: e.target.value });
+        }
+        break;
+      }
+      case "signupPwd": {
+        if (e.target.value.length < 6) {
+          this.setState({ passwordError: "minimum 6 characters needed." });
+        } else {
+          this.setState({ passwordError: "" });
+          this.setState({ password: e.target.value });
+        }
+        break;
+      }
+      case "signupPwdConfirm": {
+        if (e.target.value !== this.state.password) {
+          this.setState({ passwordConfirmError: "Mismatch" });
+        } else {
+          this.setState({ passwordConfirmError: "" });
+        }
+        break;
+      }
+      default:
+        break;
+    }
+  };
+
+  loadSignIn = () =>
+    ReactDOM.render(<SignIn />, document.getElementById("root"));
+
+  render() {
+    const { classes } = this.props;
+    return (
+      <div className={classes.signUpContainer}>
+        <div className={classes.leftSideContainer}>
+          <div className={classes.titleLogo}>
+            <Typography variant="h4">CHEF'S MENU</Typography>
+          </div>
+          <Card className={classes.card}>
+            <CardContent>
+              <Typography className={classes.title} gutterBottom>
+                Create an account
+              </Typography>
+              <form autoComplete="off">
+                <Box display="flex">
+                  <TextField
+                    className={classes.textField}
+                    required
+                    autoFocus
+                    id="signupName"
+                    name="signupName"
+                    label="Name"
+                    placeholder="Enter your name"
+                    error={this.state.usernameError !== "" ? true : false}
+                    variant="outlined"
+                    onChange={this.validateChange}
+                    helperText={this.state.usernameError}
+                  />
```

Comment:
this is also uncontrolled. Controlled input components are the official React recommendation as per link above link

Reply:
Done.